tools/plugins/com.liferay.ide.server.core/src/com/liferay/ide/server/util/PingThread.java

```diff
@@ -84,15 +84,17 @@ protected void ping() {
 
 				URL pingUrl = new URL(url);
 				URLConnection conn = pingUrl.openConnection();
+
 				((HttpURLConnection)conn).setInstanceFollowRedirects(false);
-				((HttpURLConnection)conn).getResponseCode();
+				int code = ((HttpURLConnection)conn).getResponseCode();
 
 				// ping worked - server is up
-				if (!stop) {
+				if (!stop && code == 200) {
 					Thread.sleep(200);
 					behaviour.setServerStarted();
+					stop = true;
 				}
-				stop = true;
+				Thread.sleep(1000);
```

Comment:
cause it will return 404 immediately and we consider it is a right response , so the ping thread will only run ping function one time , I make it sleep 1 second if it returns 404.
and it doesn't matter if it is in else block cause in if it will set stop to true and this will stop the while block.